internal/storage/image.go

```diff
@@ -279,6 +279,14 @@ func (svc *imageService) ListImages(systemContext *types.SystemContext, filter s
 		}
 		newImageCache := make(imageCache, len(images))
 		for i := range images {
+			// determines if an images is intermediate, and if it is, don't return it
+			isIntermediate, err := svc.isIntermediate(&images[i])
```

Comment:
nit: since we're allocating
`image := &images[i]`
below we should use allocate it above and use `image` for this call